contrib/mhchem/mhchem.js

```diff
@@ -0,0 +1,1687 @@
+﻿/* eslint-disable */
+/* -*- Mode: Javascript; indent-tabs-mode:nil; js-indent-level: 2 -*- */
+/* vim: set ts=2 et sw=2 tw=80: */
+
+/*************************************************************
+ *
+ *  MathJax/extensions/TeX/mhchem.js
+ *
+ *  Implements the \ce command for handling chemical formulas
+ *  from the mhchem LaTeX package.
+ *
+ *  ---------------------------------------------------------------------
+ *
+ *  Copyright (c) 2011-2015 The MathJax Consortium
+ *  Copyright (c) 2015-2018 Martin Hensel
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+//
+// Coding Style
+//   - use '' for identifiers that can by minified/uglified
+//   - use "" for strings that need to stay untouched
+
+// version: "3.3.0" for KaTeX
+
+
+// Add \ce, \pu, and \tripledash to the KaTeX macros.
+
+katex.__defineMacro("\\ce", function(context) {
+  return mhchem.expand(context.consumeArgs(1)[0], "ce")
+});
+
+katex.__defineMacro("\\pu", function(context) {
+  return mhchem.expand(context.consumeArgs(1)[0], "pu");
+});
+
+//  Needed for \bond for the ~ forms
+//  Raise by 2.56mu, not 2mu. We're raising a hyphen-minus, U+002D, not 
+//  a mathematical minus, U+2212. So we need that extra 0.56.
+katex.__defineMacro("\\tripledash", "{\\vphantom{-}\\raisebox{2.56mu}{$\\mkern2mu"
++ "\\tiny\\text{-}\\mkern1mu\\text{-}\\mkern1mu\\text{-}\\mkern2mu$}}");
+
+
+var mhchem = (function () {
```

Comment:
@ylemkimon My purpose here is to prevent functions like `mhchemParser()` or `texify()` from becoming available globally. The current arrangement does this successfully. It there any compelling reason to change it?

Reply:
Please be more specific. What should this line of code look like?

Reply:
I think you'll want to get rid of this line and at the bottom of the file replace
```
    return {
     expand: chemParse,
   };
```
with
```
    module.exports = {
        expand: chemParse,
    };
```
and drop the trailing:
```
})();
```
I'm curious about why `chemParse` was being exported.  Is it for debugging purposes?

Reply:
> our other code doesn't do this.

Our other code is bundled by webpack into a single file. This file remains separate and is called from a reference in an HTML <head>

> and drop the trailing `})();`

Then functions `mhchemParser()`,  `texify()`, `assertNever()`, and `assertString()` become global. Why would we want to do that?

Reply:
As usual @ylemkimon  is correct. It also turns out that `mhchem.js` does not need a `module.exports = { expand: chemParse, };` statement. `chemParse` is called locally. It's really the macro callbacks that are being exported.

I've modified the file accordingly.